parse/parser_internal_test.go

```diff
@@ -990,6 +990,7 @@ func TestParseFunctionParameter(t *testing.T) {
 	initParseFnMap()
 	tests := []struct {
 		buf         TokenBuffer
+		setupScope  setupScopeFn
 		expected    []string
 		expectedErr error
```

Comment:
I think we almost done, and last comment is plz add chkScopeFn to `TestParseFunctionParameter` to check whether symbols are successfully saved (rebase first! and you will find similar pattern in `ParseIfStatement`)

Reply:
chkScopeFn is check only current scope.
parseBlockStatement does not have entering and leaving scope feature currently.
If program <b>leaves current scope</b>, this chkScopeFn always will return false..
For example,
```go
if (true) {
   int a = 1
}
```
We want to check the value "a" is in right scope, but after parsing above, <b>current scope</b> is outer scope. so chkScopeFn will be failed.

So we should think other way.

Reply:
#290 pr created, after merge we can fix this issue!

Reply:
We have another problem!
Now, we can go to inner scope. It is same as we can search root to leaf.
But how about this example?

```go
if (a == 5) {
    int b = 3;
} else {
    int b = 4;
}
```

If we want "b" in <b>else block</b>, but problem may check <b>if block first.</b>
I think it is incorrect.